### ISSUE TITLE
getwork protocol connects, but cannot generate DAG

### ISSUE THREAD

**Issue by cesar-v** [no-fee-ethereum-mining/nsfminer#277]
I'm running a local Geth node and I'm unable to mine using an RTX3060ti over getwork://.

It refuses to generate a DAG.  I don't ever receive errors, it just continues to try work until I cancel it.

`05:15:58 miner nsfminer 1.3.14+commit.3bbc66f3.dirty (No stinkin' fees edition)`
`05:15:58 miner Copyright 2021 Jean M. Cyr, Licensed under the terms`
`05:15:58 miner  of the GNU General Public License Version 3`
`05:15:58 miner https://github.com/no-fee-ethereum-mining/nsfminer`
`05:15:58 miner Build: windows/release/msvc`
`05:15:58 miner 3rd Party: MSVC 19.16.27045.0, CUDA 11.3, Boost 1.76.0`
`05:15:58 miner 3rd Party: OpenSSL 1.1.1j  16 Feb 2021`
`05:15:58 miner Running as user: Valued Customer`
`No OpenCL platforms found`
`05:15:58 miner Configured pool 127.0.0.1:8545`
`05:15:58 miner Selected pool 127.0.0.1:8545`
`05:15:58 miner Established connection to 127.0.0.1:8545`
`05:15:58 miner Spinning up miners...`
`05:15:58 cu-0  Using Pci 0000:2d:00.0: NVIDIA GeForce RTX 3060 Ti (Compute 8.6) Memory : 8.00 GB`
`05:16:03 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:08 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:13 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:18 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:23 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:28 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:33 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:38 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:43 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:48 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:53 miner 0:00 A0 0.00 h - cu0 0.00`
`05:16:58 miner 0:01 A0 0.00 h - cu0 0.00`
`05:17:03 miner 0:01 A0 0.00 h - cu0 0.00`
`05:17:08 miner 0:01 A0 0.00 h - cu0 0.00`
`05:17:13 miner 0:01 A0 0.00 h - cu0 0.00`
`05:17:18 miner 0:01 A0 0.00 h - cu0 0.00`
`05:17:23 miner 0:01 A0 0.00 h - cu0 0.00`
`...`

My local node is synced and I'm able to mine on it using PhoenixMiner successfully.  I'd prefer to avoid those stinkin` fees if possible.

**Comment by jean-m-cyr** [comment]
Yeah, getwork is broken. Unfortunately I don't have a full node or proxy to fix it with.

**Comment by daxime** [comment]
Hi, i have the same issue. I use this miner since a month ago on Ubuntu, but since 2 days it's not working anymore ...
```
03:53:53 miner nsfminer 1.3.14 (No stinkin' fees edition)
03:53:53 miner Copyright 2021 Jean M. Cyr, Licensed under the terms
03:53:53 miner  of the GNU General Public License Version 3
03:53:53 miner https://github.com/no-fee-ethereum-mining/nsfminer
03:53:53 miner Build: linux/release/gnu
03:53:53 miner 3rd Party: GCC 10.2.0, Boost 1.76.0
03:53:53 miner 3rd Party: OpenSSL 1.1.1j  16 Feb 2021
03:53:53 miner Running as user: daxime
03:53:53 miner Configured pool eu1.ethermine.org:5555
03:53:53 miner Selected pool eu1.ethermine.org:5555
03:53:54 miner Stratum mode : Eth-Proxy compatible
03:53:54 miner Established connection to eu1.ethermine.org:5555
03:53:54 miner Spinning up miners...
03:53:54 cl-0  Using Pci 0000:0a:00.0: Navi 14 [Radeon RX 5500/5500M / Pro 5500M] (OpenCL 2.0 ) Memory : 7.98 GB
03:53:54 miner Epoch : 425 Difficulty : 4.29 Gh
03:53:54 miner Job: eb2fb651 blk: 12777760 eu1.ethermine.org:5555
03:53:55 miner Job: 58bb31bb blk: 12777760 eu1.ethermine.org:5555
03:53:57 miner Job: a070c93b blk: 12777760 eu1.ethermine.org:5555
03:53:58 cl-0  Required GPU mem: Total 4.39 GB, Cache 69.12 MB, DAG 4.32 GB, Miscellaneous 60.00 B
03:53:58 miner 0:00 A0 0.00 h - cl0 0.00
03:54:01 miner Job: 53bf06f3 blk: 12777761 eu1.ethermine.org:5555
```
After the last line, nothing more happen ...

**Comment by grenelt** [comment]
1. @cesar-v Your log doesn't indicate here that even the creating for DAG startet. Even if getwork may be broken there is maybe still another issue...
2. @daxime DAG creation starts but there is no line that it finishes. You're obviously not on the getwork - so here i'm quite sure you have another problem. Maybe too hard memory overclocking - a lot of cards doesn't cope with the hard overclocking settings you can find out there. I.e. i have three RTX3070 and none of them work with something higher then mc +1050  while cooling is quite good and gpu temperature is constantly in the mid 30s...
